include/dgl/aten/coo.h

```diff
@@ -134,6 +131,12 @@ struct COOMatrix {
            aten::IsNullArray(data);
   }
 
+  inline bool IsPinned() const {
```

Comment:
Since we set internal member `is_pinned` in the constructor, it seems odd to me this member function doesn't use that. If this is only used internally for a check, can we make it private and change the name to make clear it will initialize a cuda context? e.g. `CheckIfPinnedInCUDA()`

Reply:
Good point.
I changed the function name to `CheckIfPinnedInCUDA`. So, there are two ways to check if the coo/csr is under pinned memory:
- The 'soft' way: `is_pinned` flag. This method does not need to init. cuda context and flexible. It can be updated by the 'hard' method or by developer when memory is pinned by DGL (not pytorch).
- The 'hard/strict' way: check via CUDA runtime API. This is a strict way but involves CUDA context. I use it to update the flag, only if necessary (when the pinned memory pool is not controlled by DGL and we need this sanity check on DGL side).